### PR TITLE
don't preload metadata for the sub-childen

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes to broad preloading logic introduced with https://github.com/nextcloud/server/pull/39694

- Only preload when `Depth = 1`, this stops preloading for the children of sub-folders
- Don't trigger preload logic when no metadata is requested.

### CODE REVIEW

**Review by artonge — APPROVED** [review]

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 1]
```diff
@@ -465,13 +465,17 @@ public function handleGetProperties(PropFind $propFind, \Sabre\DAV\INode $node)
...
+			$requestedMetaData = [];
+			foreach ($requestProperties as $requestProperty) {
+				if (isset(self::ALL_METADATA_PROPS[$requestProperty])) {
+					$requestedMetaData[] = self::ALL_METADATA_PROPS[$requestProperty];
 				}
+			}
+			if (
+				$this->config->getSystemValueBool('enable_file_metadata', true) &&
+				$propFind->getDepth() === 1 &&
+				$requestedMetaData
```
Comment (ArtificialOwl) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php:477]: any reason for this 3rd condition ?

**Review by ArtificialOwl — APPROVED** [review]

**Review by Altahrim — APPROVED** [review]

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php (hunk 1 above)
Comment (icewind1991) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:477]: it allows skipping the logic if no metadata fields are requested

Comment (ArtificialOwl) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:477]: but it is stays `true` even if array is empty, right ?

Comment (ArtificialOwl) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:477]: nevermind, it seems that an empty array will returns `false`

**Comment by pulsejet** [comment]
This patch cut down my SQL requests to half 👀
Should've been labelled Performance 🚀😉

**Pull request merged**